### PR TITLE
allow multiple API/Gateway addresses

### PR DESCRIPTION
**Pull request by Stebalien**
Alternative to #6 that doesn't require a migration. This is probably a terrible idea.

### CODE REVIEW

**Comment by lgierth** [comment]
I have a hunch we should really make fs-repo-migrations easier to write, with generators or templates, or something like that. Given that one of the most prominent devs is trying to avoid migrations :)

**Comment by Stebalien** [comment]
My primary issue with migrations is having to *download* them. That adds an extra update step that can easily fail.

Also, *this* change will probably break a ton of our test tooling (as `ipfs config --json "Addresses.API" '[...]'`).

**Comment by Stebalien** [comment]
So, @whyrusleeping at least doesn't think this is that bad and, unlike #6, this doesn't break any existing tooling (i.e., anything that runs `ipfs config --json "Addresses.API" "/ip4/..."`).

Does this look reasonable @magik6k @lgierth @overbool?

File: types.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	}
+	*o = []string{value}
+	return nil
+}
+
+// MarshalJSON conforms to the json.Marshaler interface.
+func (o Strings) MarshalJSON() ([]byte, error) {
+	switch len(o) {
+	case 0:
+		return json.Marshal(nil)
```
Comment (magik6k) [inline types.go]: Shouldn't this be returning an empty array? Or will the marshaller just omit the entry if it's nil?

Comment (Stebalien) [reply types.go]: Exactly. Basically, I'd like to use `null` for the case where *no* API addresses are specified.

Comment (Stebalien) [reply types.go]: @overbool want to file a PR against go-ipfs that uses this?

Comment (overbool) [reply types.go]: @Stebalien Did you mean this https://github.com/ipfs/go-ipfs/issues/5401?

**Pull request merged**